Hogajama/hogajama-domain/src/main/java/com/gepardec/hogarama/domain/unitmanagement/cache/ActorCache.java

```diff
@@ -0,0 +1,56 @@
+package com.gepardec.hogarama.domain.unitmanagement.cache;
+
+import com.gepardec.hogarama.domain.unitmanagement.dao.ActorDAO;
+import com.gepardec.hogarama.domain.unitmanagement.entity.Actor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class ActorCache {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ActorCache.class);
+
+    @Inject
+    private ActorDAO dao;
+
+    private Map<String, Actor> cache = new HashMap<>();
+
+    public void invalidateCache(@Observes Actor actor) {
+        LOG.info("Invalidate actor cache.");
+        cache = new HashMap<>();
+    }
+
+    public Optional<Actor> getByDeviceId(String deviceId) {
+        Optional<Actor> actor = getCachedActor(deviceId);
+        if ( actor.isPresent()) {
+            return actor;
+        }
+        return loadActor(deviceId);
+    }
+
+    private synchronized Optional<Actor> loadActor(String deviceId) {
+        Optional<Actor> optionalActor = getCachedActor(deviceId);
```

Comment:
Is this not fix empty? Because you already have the same check in the above method, that calls this one. So you call this methon only if actor.isPresent() == false

Reply:
@eerofeev If several threads come at the same time, they pass the first check and wait at the synchronized method. The first loads the actor, without the second check, the other threads would load it again.